pallets/aleph/src/lib.rs

```diff
@@ -123,6 +129,18 @@ pub mod pallet {
     pub(super) type FinalityScheduledVersionChange<T: Config> =
         StorageValue<_, VersionChange, OptionQuery>;
 
+    #[pallet::hooks]
+    impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+        fn on_finalize(block_number: T::BlockNumber) {
+            if block_number % DEFAULT_SESSION_PERIOD.into() == (DEFAULT_SESSION_PERIOD - 1).into() {
+                <frame_system::Pallet<T>>::deposit_log(DigestItem::Consensus(
```

Comment:
I'm wondering how does it actually work. I.e. is this log somehow verified by the remaining validators :P? I would assume it is, but I guess it would be good to make sure, because nobody here is familiar with how these logs work...

Reply:
Actually, somebody has a raw idea how they work :P The check is done in some convoluted way, i.e. other validators don't trust logs (they are not part of prehash for example). They execute the block and after executing it they check if the logs they got from execution (particularily from `on_finalize`) match the ones that was send with the block. Checks:
https://github.com/paritytech/substrate/blob/master/frame/executive/src/lib.rs#L588
https://github.com/paritytech/substrate/blob/master/frame/executive/src/lib.rs#L594